aten/src/ATen/native/SoftMax.cpp

```diff
@@ -131,7 +131,18 @@ void host_softmax(
     Tensor output,
     const Tensor& input,
     const int64_t dim,
-    bool* mask = nullptr) {
+    bool* mask = nullptr,
+    const c10::optional<int64_t> mask_type_ = NULL) {
+
+  if (MaskedSoftMax) {
+    TORCH_CHECK(mask_type_.has_value(), "Mask Type should be defined");
+    int64_t mask_type = mask_type_.value();
+    TORCH_CHECK((mask_type == 0) || (mask_type == 1), "Mask Type should be 0 (src_mask) or 1 (src_key_padding_mask)");
```

Comment:
Maybe re-org this part of logic a bit? since mask_type == 0 does not work on CPU

Reply:
Since adding support for mask_type == 0 is in progress I chose to have that check separate as it will be removed soon